### PR TITLE
added option to save steps of iteration

### PR DESCRIPTION
**Pull request by KirillZubov**

### CODE REVIEW

File: src/pde_solve_ns.jl [hunk 1]
```diff
@@ -69,13 +70,16 @@ function DiffEqBase.solve(
         mean(sum(abs2, g(X.data) - u) for (X,u) in predict_n_sde())
     end
 
+    iters = Float32[]
```
Comment (ChrisRackauckas) [inline src/pde_solve_ns.jl]: eltype(X0)

**Comment by ChrisRackauckas** [comment]
add to the other method as well?

**Comment by KirillZubov** [comment]
> add to the other method as well?

ok

File: src/pde_solve_ns.jl (hunk 1 above)
Comment (KirillZubov) [reply src/pde_solve_ns.jl]: In pde_solve tests x0 is Array of Float64. I suppose, it should be eltype(u0(X0)[1].data) but it is always Array of Float32.

Comment (ChrisRackauckas) [reply src/pde_solve_ns.jl]: You should make the inputs in the test be Float32 then for type stability

Comment (KirillZubov) [reply src/pde_solve_ns.jl]: ok

**Pull request merged**